### PR TITLE
Make token-verification a REST endpoint

### LINKED ISSUE #1026: Making token verification a REST endpoint
RFID server uses token functions to verify user membership before creating or deleting an RFID. Token-function uses passport.js which is located in `main_endpoints/util`. https://github.com/SCE-Development/Core-v4/blob/d0eb4fadc91da5f939c2b03613d81107cc24d161/api/peripheral_api/routes/RFID.js#L7 https://github.com/SCE-Development/Core-v4/blob/d0eb4fadc91da5f939c2b03613d81107cc24d161/api/util/token-functions.js#L6
Problem is that, when in production, `peripheral_api/*` has no way to access `main_endpoints/*` which will break production.  Making token verification an endpoint will allow usage of token verification in peripheral server

Use the endpoint in Auth.js for token verification and modify API tests to adapt to the new changes.

### PR DESCRIPTION
**Pull request by neelp03**
Resolves #1026

### CODE REVIEW

File: api/util/passport.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const JwtStrategy = require('passport-jwt').Strategy;
 const ExtractJwt = require('passport-jwt').ExtractJwt;
-const User = require('../models/User');
-const { secretKey } = require('../../config/config.json');
+const User = require('../main_endpoints/models/User');
```
Comment (evanugarte) [inline api/util/passport.js:3]: this is inside main_endpoints, meaning we will still get the below error in docker:
image

we have two options:
1. move the user schema to util/
2. make token verification a REST endpoint (`/api/verifyToken`) for other services outside of `main_endpoints` to call on.

1 would be much easier than 2. however the schema would be in an uncommon place. your call.

**Review by evanugarte — APPROVED** [review]
Tested locally and it works. This is very useful for #1004 and #1015

File: api/util/passport.js (hunk 1 above)
Comment (neelp03) [reply api/util/passport.js]: option 2 is the move since it will help later with other token verification in peripheral server

**Review by andrewtedi — APPROVED** [review]
Great work Neel!

**Pull request merged**